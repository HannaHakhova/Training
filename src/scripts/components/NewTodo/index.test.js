import React from 'react';
import {shallow, mount, add} from 'enzyme';
import NewTodo from './index';

const container = shallow(<NewTodo createTask={() => {}} />);

describe('NewTodo component', () => {
    it('should match expected snapshot', () => {
        expect(container).toMatchSnapshot();
    });
});

it('creates task on the button click', () => {
    container.setState({value: 'to do task1'});
    container.find('button').simulate('click');
    expect(container.state().value).toEqual('');
});

it('handles change on the input', () => {
    container.find('input').simulate('change', {target: {value: 'testValue'} });
    expect(container.state().value).toEqual('testValue');
});
