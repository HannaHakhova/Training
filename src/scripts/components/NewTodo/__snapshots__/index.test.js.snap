// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodo component should match expected snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="heading"
  >
    New To Do List
  </h1>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="button button2"
    disabled={true}
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
